src/funfuzz/js/shell_flags.py

```diff
@@ -24,165 +22,290 @@
 
 
 @lru_cache(maxsize=None)
-def shellSupportsFlag(shellPath, flag):  # pylint: disable=invalid-name,missing-docstring,missing-return-doc
-    # pylint: disable=missing-return-type-doc
-    return inspect_shell.shellSupports(shellPath, [flag, '-e', '42'])
+def shell_supports_flag(shell_path, flag):
+    """Returns whether a particular flag is supported by a shell.
+
+    Args:
+        shell_path (str): Path to the required shell.
+        flag (str): Intended flag to test.
 
+    Returns:
+        bool: True if the flag is supported, i.e. does not cause the shell to throw an error, False otherwise.
+    """
+    return inspect_shell.shellSupports(shell_path, [flag, "-e", "42"])
 
-def chance(p):  # pylint: disable=invalid-name,missing-docstring,missing-return-doc,missing-return-type-doc
-    return random.random() < p
 
+def chance(i, always=False):
+    """Returns a random boolean result based on an input probability.
 
-def randomFlagSet(shellPath):  # pylint: disable=invalid-name,missing-param-doc,missing-return-doc
-    # pylint: disable=missing-return-type-doc,missing-type-doc,too-complex,too-many-branches,too-many-statements
-    """Return a random list of CLI flags appropriate for the given shell.
+    Args:
+        i (float): Intended probability.
+        always (bool): Causes the function to always return True. For testing purposes.
 
-    Only works for spidermonkey js shell. Does not work for xpcshell.
+    Returns:
+        bool: Result based on the input probability, unless the "always" parameter is set.
     """
-    args = []
+    return (random.random() < i) if not always else True
```

Comment:
I like this idea, but it scatters test code throughout (if you forget to pass-through `always` somewhere, for example).

Another way would be to use the [monkeypatch](https://docs.pytest.org/en/latest/monkeypatch.html) fixture to override the `chance` function during testing, like this:

    def test_something(monkeypatch):
        monkeypatch.setattr(funfuzz.js.shell_flags, 'chance', lambda _prob: True)
        # run tests here ... chance() will always return True

I've done this already in the FuzzManager tests. It keeps the code-under-test from caring that it's being tested.